lib/phoenix_live_view/html_parser.ex

```diff
@@ -0,0 +1,453 @@
+defmodule Phoenix.LiveView.HTMLParser do
+  @moduledoc false
```

Comment:
I also have a use case for parsing heex template code in a side project, and if these behaviors are now public, I think a brief moduledoc makes sense.

Reply:
I'm not so sure. The way the template AST is generated for the formatter is really quite janky and the HTMLAlgebra generator does not even try to format [every expression](https://github.com/phoenixframework/phoenix_live_view/blob/v0.20.14/lib/phoenix_live_view/html_algebra.ex#L483-L486) (because block expressions are incomplete, they are unable to be parsed with `Code.string_to_quoted`), so it's not really a "proper" AST and still requires extra massaging to work with.

If the AST generated included quoted versions of the block expressions, then I would agree that it should be properly exposed.